homeassistant/components/trafikverket_camera/__init__.py

```diff
@@ -30,3 +38,33 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    # Change entry unique id from location to camera id
+    if entry.version == 1:
+        location = entry.data[CONF_LOCATION]
+        api_key = entry.data[CONF_API_KEY]
+
+        web_session = async_get_clientsession(hass)
+        camera_api = TrafikverketCamera(web_session, api_key)
+
+        try:
+            camera_info = await camera_api.async_get_camera(location)
+        except Exception:  # pylint: disable=broad-except
+            return False
```

Comment:
I suggest adding logging when we return False, as otherwise, the user wonders why the migration failed.

Reply:
Clever. I'll add that. 👍